File: INTEGRACOES/scripts/content_bridge.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import dotenv from 'dotenv';
import { GoogleGenerativeAI } from "@google/generative-ai";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Load environment variables from .env
dotenv.config({ path: path.join(__dirname, '..', '..', '.env.local') });

/**
 * THE BRIDGE (STABLE VERSION) - Amém IA Content Automation Engine
 * Uses Official Google SDK for maximum reliability.
 */
class AmemIABridge {
    constructor() {
        this.key = process.env.GEMINI_API_KEY;
        this.baseUrl = "https://generativelanguage.googleapis.com/v1beta/models/gemini-2.5-flash:generateContent";
        
        this.basePath = "G:\\EMPRESA_DIGITAL\\AMEM_IA\\MARKETING";
        this.scriptPath = path.join(this.basePath, "ROTEIROS");
        this.imagePath = path.join(this.basePath, "IMAGENS");
        
        if (!fs.existsSync(this.scriptPath)) fs.mkdirSync(this.scriptPath, { recursive: true });
        if (!fs.existsSync(this.imagePath)) fs.mkdirSync(this.imagePath, { recursive: true });
    }

    async generateViralScript(theme) {
        console.log(`[*] Generating viral script for theme: ${theme}...`);
        
        const prompt = `Você é Brian, o Mega Brain da Amém IA.
        OBJETIVO: Criar um roteiro de vídeo curto (Reels/TikTok) ultra-persuasivo.
        TEMA: "${theme}"
        PÚBLICO: Cristãos que buscam profundidade espiritual.
        
        O roteiro deve ter:
        1. HOOK: Gancho brutal para os primeiros 3 segundos.
        2. CONTENT: Conexão emocional profunda (storytelling).
        3. VISUALS: Sugestão de imagens/cenas para o B-Roll.
        4. CTA: Direcionamento imediato para o app amemia.com.br.
        
        Retorne APENAS um JSON puro no formato:
        { "hook": "...", "content": "...", "visuals": "...", "cta": "..." }`;

        try {
            console.log(`[DEBUG] Fetching from: ${this.baseUrl}`);
            const response = await fetch(`${this.baseUrl}?key=${this.key}`, {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({
                    contents: [{ parts: [{ text: prompt }] }]
                })
            });

            const data = await response.json();
            if (data.error) {
                console.error("[!] API Error:", data.error);
                return null;
            }
            if (!data.candidates || !data.candidates[0].content) {
                console.error("[!] Unexpected API structure:", JSON.stringify(data, null, 2));
                return null;
            }
            const rawText = data.candidates[0].content.parts[0].text;
            const jsonMatch = rawText.match(/\{[\s\S]*\}/);
            return JSON.parse(jsonMatch ? jsonMatch[0] : rawText);
        } catch (error) {
            console.error("[!] Error in Gemini Generation:", error);
            return null;
        }
    }

    async generateNotebookLMBriefing(topic) {
        console.log(`[*] Preparing NotebookLM research briefing for: ${topic}...`);
        const prompt = `Como Brian, compile os fundamentos teológicos mais profundos sobre "${topic}" para servir de base no NotebookLM. 
        Inclua: 1. Contexto Histórico. 2. Raízes no Grego/Hebraico. 3. Aplicação Prática Moderna. 4. Referências Cruzadas Supremas.`;

        try {
            const response = await fetch(`${this.baseUrl}?key=${this.key}`, {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify({
                    contents: [{ parts: [{ text: prompt }] }]
                })
            });

            const data = await response.json();
            const text = data.candidates[0].content.parts[0].text;
            const filePath = path.join(this.basePath, "ESTUDOS", `${topic.replace(/\s+/g, '_')}_briefing.txt`);
            if (!fs.existsSync(path.dirname(filePath))) fs.mkdirSync(path.dirname(filePath), { recursive: true });
            fs.writeFileSync(filePath, text);
            console.log(`[📓] NotebookLM Briefing saved: ${filePath}`);
            return text;
        } catch (error) {
            console.error("[!] Error in Research Generation:", error);
            return "";
        }
    }

    saveScript(name, script) {
        const textPath = path.join(this.scriptPath, `${name}.txt`);
        const csvPath = path.join(this.imagePath, "canva_bulk_import.csv");
        
        const textContent = `==== ROTEIRO PARA CAPCUT (SCRIPT TO VIDEO) ====
TÍTULO: ${script.hook}
SUGESTÃO VISUAL: ${script.visuals}

TEXTO PARA NARRAÇÃO:
${script.content}

CTA FINAL:
${script.cta}
==============================================`;

        fs.writeFileSync(textPath, textContent);
        
        // Append to Canva CSV for Bulk Create
        // Format: Title, VisualHint, Body, CTA
        const csvLine = `"${script.hook.replace(/"/g, '""')}","${script.visuals.replace(/"/g, '""')}","${script.content.replace(/"/g, '""')}","${script.cta.replace(/"/g, '""')}"\n`;
        if (!fs.existsSync(csvPath)) {
            fs.writeFileSync(csvPath, "Title,VisualHint,Body,CTA\n");
        }
        fs.appendFileSync(csvPath, csvLine);
        
        console.log(`[+] Script saved to: ${textPath}`);
        console.log(`[+] Canva CSV updated for Bulk Create: ${csvPath}`);
    }

    async bulkContentGeneration(themes) {
        console.log(`[🔥] STRATEGIC BULK RUN: Starting ${themes.length} generations...`);
        for (let i = 0; i < themes.length; i++) {
            const name = `bulk_criativo_${String(i+1).padStart(3, '0')}`;
            await this.run(themes[i], name);
            // NotebookLM Briefing for each themes as well
            await this.generateNotebookLMBriefing(themes[i]);
        }
    }

    async run(theme, name) {
        const script = await this.generateViralScript(theme);
        if (script) {
            this.saveScript(name, script);
            console.log(`[🚀] SUCCESS: Content '${name}' automated and ready for Canva/CapCut!`);
        }
    }
}

export async function bulkContentGeneration(themes) {
    const bridge = new AmemIABridge();
    console.log(`[🔥] STRATEGIC BULK RUN: Starting ${themes.length} generations...`);
    
    let canvaCSV = "Title,VisualHint,Body,CTA\n";
    let capcutScripts = "";
    let notebookLMBriefing = "";

    for (let i = 0; i < themes.length; i++) {
        const name = `bulk_criativo_${String(i+1).padStart(3, '0')}`;
        const script = await bridge.generateViralScript(themes[i]);
        if (script) {
            // Canva
            canvaCSV += `"${script.hook.replace(/"/g, '""')}","${script.visuals.replace(/"/g, '""')}","${script.content.replace(/"/g, '""')}","${script.cta.replace(/"/g, '""')}"\n`;
            
            // CapCut
            capcutScripts += `\n==== ROTEIRO: ${script.hook} ====\nVISUAIS: ${script.visuals}\nTEXTO: ${script.content}\nCTA: ${script.cta}\n========================\n`;

            // NotebookLM (Accumulate or return the last one as briefing example)
            const briefing = await bridge.generateNotebookLMBriefing(themes[i]);
            notebookLMBriefing += `\n# ESTUDO: ${themes[i]}\n${briefing}\n---\n`;
        }
    }

    return { canvaCSV, capcutScripts, notebookLMBriefing };
}

File: INTEGRACOES/scripts/content_bridge.py
import os
import requests
import json

# THE BRIDGE - Amém IA Content Automation Engine
# Connects Gemini, ElevenLabs, and Canva for seamless viral production.

class AmemIABridge:
    def __init__(self):
        # Professional Environment Loading
        self.gemini_key = os.getenv("GEMINI_API_KEY")
        self.elevenlabs_key = os.getenv("ELEVENLABS_API_KEY")
        self.voice_id = os.getenv("ELEVENLABS_VOICE_ID", "pNInz6obpg9zX77S5y5o") # Default Pastoral Voice
        self.canva_api_key = os.getenv("CANVA_API_KEY")
        
        self.base_path = r"G:\EMPRESA_DIGITAL\AME_IA\MARKETING"
        self.audio_path = os.path.join(self.base_path, "AUDIOS")
        self.video_path = os.path.join(self.base_path, "VIDEOS")

    def generate_script(self, theme):
        """Uses Gemini to refine the script into viral segments."""
        print(f"[*] Generating viral script for theme: {theme}...")
        # Logic to call Gemini API and return structured JSON
        return {
            "hook": "Você não é um erro.",
            "content": "Deus não faz rascunhos. Ele faz obras primas.",
            "cta": "Link na Bio."
        }

    def generate_voice(self, text, output_name):
        """Uses ElevenLabs to generate the Pastor IA voice with FREE fallback."""
        if not self.elevenlabs_key:
            print("[!] ElevenLabs API key missing. Saving script as text for CapCut TTS fallback.")
            text_path = os.path.join(self.base_path, "ROTEIROS", f"{output_name}.txt")
            os.makedirs(os.path.dirname(text_path), exist_ok=True)
            with open(text_path, "w", encoding="utf-8") as f:
                f.write(text)
            print(f"[+] Text saved for manual narrating: {text_path}")
            return text_path

        print(f"[*] Generating voiceover with ElevenLabs...")
        # ... (rest of the API call logic)

    def sync_to_canva(self, content):
        """Pushes data to Canva via API or generates design prompts."""
        print(f"[*] Syncing keywords to Canva Design Engine...")
        # Canva API integration to create/update templates
        pass

    def run_full_pipeline(self, theme, name):
        script = self.generate_script(theme)
        voice_file = self.generate_voice(script['content'], name)
        print(f"\n[🚀] SUCCESS: '{name}' is ready for CapCut Pro import!")
        print(f"      - Script: {script}")
        print(f"      - Audio: {voice_file}")

if __name__ == "__main__":
    bridge = AmemIABridge()
    # Example run:
    # bridge.run_full_pipeline("Propósito Divino", "devocional_001")
